src/rust/engine/process_execution/src/remote_tests.rs

```diff
@@ -2245,6 +2248,7 @@ fn remote_workunits_are_stored() {
       },
       span_id: String::from("ignore"),
       parent_id: None,
+      display_info: None,
```

Comment:
It's not clear how a workunit name would be different from its `display_info`. If it's actually necessary, a comment in here that explains the difference between the two would be good.

Reply:
@stuhood I don't have a problem with making this `display_info` field be the same thing as its `name`, and getting rid of the logic that sets `let node_name = format!("{}", self);` on `NodeKey`. This would entail either changing the `name` field to be Optional, since not all rules are named, or keeping it as it is and simply not creating a `WorkUnit` if there's no defined `name`. Tagging @cattibrie @illicitonion @ity  since this would affect the current zipkin reporting.

Reply:
Well, see the first bullet in the linked ticket: a big question is whether we want to render rules that aren't named: I expect that the answer is no.

Reply:
It's a bit hard to judge from this commit if this is desired or not because no-one calls `add_display_info` yet, so that makes it difficult to assess if all the added state pays for itself or not. For instance, if the display_info ended up being something we could compute from the name and the rest of the state we've already got, it may make sense to compute it more locally in rust and avoid state proliferation.

If you could add the commits that use the display_info, that would help provide more useful feedback.

If there is a way to use the `name` in some way, I'm all for that because less state == better code.

Reply:
@pierrechevalier83 right now I'm inclined to go with @stuhood 's suggestion that was also raised in https://github.com/pantsbuild/pants/issues/7907 - giving a name to an `@rule` in python is the thing that controls whether or not a workunit is even recorded (that is, whether we create a `WorkUnit` struct and then call `workunit_store().add_workunit()` on it. Obviously, this would drastically reduce the amount of information that was reported in zipkin, at least until someone went through and manually added `name=<something>` to a bunch of `@rule`s in python, so I don't want to do this without checking with you and other people currently using zipkin. But if the point of named rules is to control what rules we do and do not care about reporting information about, then I think this behavior is exactly what we want. If we went this route, I would get rid of `display_info` as a separate field on `WorkUnit`, and instead write the string from python into the existing `name` field.

Another option is that I could get rid of `display_info` as a separate field in this commit set, and instead have the `name` field on `WorkUnit` preferentially get its name from the `name=` kwarg in python on an `@rule`, and just fall back to the current behavior (which is `let node_name = format!("{}", self);` on line 1216 of `nodes.rs`) if there is no name from python available. If we do this, then naming an `@rule` in python would affect the name of a WorkUnit as reported to zipkin and/or the python-side async workunit handler, but it wouldn't affect whether or not that reporting happened. If we go this route, we would probably want to also add a separate kwarg to `@rule`s that _did_ control this reporting - maybe something like `@rule(reporting=False)\ndef my_rule(a: A) -> B:`. Whether this proposed kwarg would default to `True` or `False` also depends on what behavior the people currently using zipkin want to see.

Reply:
Another consideration that was raised in https://github.com/pantsbuild/pants/issues/7907 is @blorente 's 2nd bullet point about `@rule`s defining their string representation in a more sophisticated way than just passing a static string from python. I didn't try to implement this in this draft PR because I'm not quite sure how to do it, and I think that it's fine to start with just a static string and make the python-side functionality more sophisticated later. 

I think the best way to implement this would be to continue to let `WorkUnit` have a static `String` `name` field, but make the new `get_display_info()` method on `Task` either return a static string from the `tasks::Task` struct *or* generate a `String` using the `params` field. I'm not sure how to best implement this function in python - it's probably difficult and possibly undesirable to pass a "display" function defined in python as a kwarg to a rule. Maybe we should invent our own small string formatting language that just knows how to interpolate rule parameters into a final name?

Reply:
Thanks for taking the time to lay down your thoughts here.

> [...] Obviously, this would drastically reduce the amount of information that was reported in zipkin, at least until someone went through and manually added name=<something> to a bunch of @rules in python, so I don't want to do this without checking with you and other people currently using zipkin.

Just to be explicit: I think it makes sense as a mechanism to avoid zipkin spamming. It's a decent ways to give rule authors some agency on what's worth printing or not.

> If we went this route, I would get rid of display_info as a separate field on WorkUnit, and instead write the string from python into the existing name field.

Yes, sounds good. Thanks for your last commit where you remove it. We can always re-introduce it later if we change our mind.

Reply:
@pierrechevalier83 okay in that case I'll add a commit to this PR that actually does remove a Workunit from reporting - either to the existing zipkin code or my new async workunits code - if there is no name specified on the rule.